x/ibc-rate-limit/rate_limit.go

```diff
@@ -0,0 +1,96 @@
+package ibc_rate_limit
+
+import (
+	"encoding/json"
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/ibc-go/v3/modules/core/exported"
+	"github.com/osmosis-labs/osmosis/v11/x/ibc-rate-limit/types"
+)
+
+func CheckRateLimits(ctx sdk.Context, wasmKeeper *wasmkeeper.Keeper,
+	msgType, contract string,
+	channelValue sdk.Int, sourceChannel, denom string,
+	amount string,
+) error {
+	contractAddr, err := sdk.AccAddressFromBech32(contract)
+	if err != nil {
+		return err
+	}
+
+	sendPacketMsg, _ := BuildWasmExecMsg(
+		msgType,
+		sourceChannel,
+		denom,
+		channelValue,
+		amount,
+	)
+
+	contractKeeper := wasmkeeper.NewDefaultPermissionKeeper(wasmKeeper)
```

Comment:
I think we should be passing in this contract keeper into the module, at the app wiring layer

Reply:
good call! that way we only run this once.

Reply:
changed in the `ibc-rate-limit` branch